SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -1565,6 +1565,7 @@ void CEulerSolver::CommonPreprocessing(CGeometry *geometry, CSolver **solver_con
 
 void CEulerSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container, CConfig *config, unsigned short iMesh,
                                  unsigned short iRKStep, unsigned short RunTime_EqSystem, bool Output) {
+  config->SetGlobalParam(config->GetKind_Solver(), RunTime_EqSystem);
```

Comment:
This breaks the "switching between explicit and implicit dynamically" I was telling you about.
Because we are setting a "custom" explicit mode before calling the solver in CNewtonIntegration::ComputeResiduals and in CMultiGridIntegration::MultiGrid_Cycle, but then the solver discards it.
But at the same time I like your solution as it allows base solver classes to do the right thing using the general MUSCL, Limiter, etc., and opens room for us to simplify the FluidIteration...

Maybe let's remove these SetGlobalParam from flow solvers (that may use MG or NK) for now while I think of something, but keep them in scalar solvers to allow the restarts to work properly.

Reply:
👍 57a87e5 removes the flow solver instances. What about the SetGlobalParam CDiscAdjSolver, do you foresee any issues there